### PR TITLE
Extract refresh support

### PR DESCRIPTION
**Pull request by RussTheAerialist**
Okay, that's better.

### CODE REVIEW

File: samples/refresh_tasks.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+####
+# TODO
```
Comment (RussTheAerialist) [inline samples/refresh_tasks.py]: dang it, I need to write the documentation for this.  Coming soon

File: tableauserverclient/models/schedule_item.py [hunk 2]
```diff
@@ -33,6 +33,10 @@ def __init__(self, name, priority, schedule_type, execution_order, interval_item
         self.priority = priority
         self.schedule_type = schedule_type
 
+    def __repr__(self):
+        print(self.__dict__)
```
Comment (t8y8) [inline tableauserverclient/models/schedule_item.py]: Print?

File: tableauserverclient/models/task_item.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        parsed_response = ET.fromstring(xml)
+        all_tasks_xml = parsed_response.findall(
+            './/t:task/t:extractRefresh', namespaces=NAMESPACE)
+
+        all_tasks = (TaskItem._parse_task(x) for x in all_tasks_xml)
+
+        return list(all_tasks)
+
+    @classmethod
+    def _parse_task(cls, element):
```
Comment (t8y8) [inline tableauserverclient/models/task_item.py]: _parse_element

**Comment by ferkaz** [comment]
Hey @RussTheAerialist 

This looks great, thanks for working on the extract refresh feature!

Is this safe to be used in production already? If not, do you have any ETA?

Thanks again!

**Comment by RussTheAerialist** [comment]
Hi @ferkaz,  this feature is slated to be a future release of the Tableau Server.  It's not guaranteed in the next release, but should be there.  Oh, and now that I look at it, I forgot to tag them with the rest api version, so I need to go a new pull request for that!

**Comment by ferkaz** [comment]
Sounds good, thank you!

**Comment by dungkr89** [comment]
Hi all, I tried to run the file and got the error: "refresh_tasks.py: error: too few arguments"  when I called:
python samples\refresh_tasks.py -s "http://server.it" -u "my name" -S "Default"
Any idea? Thanks

**Comment by RussTheAerialist** [comment]
Can you give us the full call stack that happens?  Also, which version of
TSC are you using?

-Russell

**Comment by dungkr89** [comment]
@RussTheAerialist  Thank you for your reply, I'm using TSC0.5.1 and the error looks like that:

capture 1

**Comment by RussTheAerialist** [comment]
Oh! okay, I see what you mean.  So the refresh tasks sample has three "commands" built into it.  You can list all of the refresh tasks, you can run one of the refresh tasks, or you get information about the refresh task.  In your case, if you run it like the following, it should work:

```
python samples\refresh_tasks.py -s "http://server.it" -u "my name" -S "Default" list
```

list will output a list of tasks, and you can then call info with the id of the task to get more details about it.  Once you've identified which one you want to run, you can then use the run command to kick it off.

**Comment by dungkr89** [comment]
@RussTheAerialist Thank you, it works well, but I guest error401001: Signin Error, Error signing in to Tableau Server. 
However I'm trying to write code to refresh a datasource on Tableau Server from a location file. It looks like use "Tableau Data Extract Command-Line Utility to refresh extract" to extract data. You have any suggest? Thank you

**Comment by RussTheAerialist** [comment]
If you are getting a sign in error either you typed the password wrong, or you don't have access to the site you are trying to login to.  If you are logging into the default site, you don't need to specify it on the command line.  Try removing the `-S "Default"`.

**Comment by tarkah** [comment]
asdf

**Comment by andylam1989** [comment]
Thank you @RussTheAerialist, I have moved to other project so I havent seen for some days. I have just tested it and it work as your suggestion. It's very nice experience, thank you again ;)

**Pull request merged**